libs/wingc/src/type_check.rs

```diff
@@ -1508,6 +1505,88 @@ impl<'a> TypeChecker<'a> {
 		}
 	}
 
+	fn pseudo_reify_std_class(&mut self, env: &SymbolEnv, original_type_name: &str, type_param: TypeRef) -> TypeRef {
+		let std_lookup = env.try_lookup("std", None).unwrap();
+		let std_namespace = &std_lookup.as_namespace().unwrap().env;
+		let original_type = std_namespace
+			.try_lookup(original_type_name, None)
+			.unwrap()
+			.as_type()
+			.unwrap();
```

Comment:
Lets add appropriate class names as constants in _lib.rs_: `WINGSDK_ARRAY` or `STD_ARRAY` set to `"std.Array"` and then pass the dot delimited string to this function and use `env.netsted_lookup_str()`.